### PR TITLE
Allow Uploadable annotation in parent classes

### LINKED ISSUE #793: Allow to Put @Vich\UploadableField Annotation into a Parent Class
Using the `@Vich\UploadableField` annotation in a parent class (`@ORM\MappedSuperclass`) does not work, since the annotation driver only checks the class itself for the `UploadableField` annotation.

So you have to clone the `UploadableField` and its methods into each derivation class.

The following modification in the AnnotationDriver.php loadMetadataForClass() helps:

old code: 

`foreach ($class->getProperties() as $property) {`

new code:

~~~
$classes = array();
do {
    $classes[] = $class;
    $class = $class->getParentClass();
} while (false !== $class);
$classes = array_reverse($classes, false);
$properties = array();
foreach ($classes as $class) {
    $properties = array_merge($properties, $class->getProperties());
}
foreach ($properties as $property) {
~~~
Note:
- loop borrowed from `vendor/jms/metadata/src/Metadata/MetadataFactory.php::getClassHierarchy`
- Actually the code can be optimized to avoid using two loops.
- I'm referring to version `1.4.2`, the code has changed on trunk on github but has still the  same behavior.

### PR DESCRIPTION
**Pull request by etshy**
Closes #793

This PR allow the bundle to search for the Uploadable Annotation in parent Classes.

Thanks to HSteeb for the code

### CODE REVIEW

File: Metadata/Driver/AnnotationDriver.php [hunk 1]
```diff
@@ -19,36 +19,47 @@ class AnnotationDriver implements AdvancedDriverInterface
      * @deprecated
      */
     const UPLOADABLE_ANNOTATION = Uploadable::class;
-
+
```
Comment (garak) [inline Metadata/Driver/AnnotationDriver.php]: This is bad. Please refer to [StyleCI issues](https://github.styleci.io/analyses/zGAaOl) and fix

Comment (etshy) [reply Metadata/Driver/AnnotationDriver.php]: I guess that's the default behaviour in PHPStorm as I don't remember configuring that
I'll change that as soon as I can.

File: Tests/DummyFile.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
+<?php
+declare(strict_types=1);
```
Comment (etshy) [inline Tests/DummyFile.php]: I saw an error in [StyleCI](https://github.styleci.io/analyses/zGAaOl) about this line but I can't see what's wrong exactly.
(the - and + line seems to be exactly the same).
Should I delete this line ?

Comment (garak) [reply Tests/DummyFile.php]: It looks like there's an extra empty line after `declare` line. Anyway, I don't see a reason to declare strict types in such file

Comment (etshy) [reply Tests/DummyFile.php]: It's just creating by default with my phpstorm settings.
It's true  that in tests file it's not needed

**Comment by etshy** [comment]
Hmm I'm not sure what to do about the last styleci error.
I tried to add or remove empty line/indents but it always fails.

**Comment by garak** [comment]
> Hmm I'm not sure what to do about the last styleci error.
> I tried to add or remove empty line/indents but it always fails.

You're missing an empty line at the end of file

**Comment by garak** [comment]
Thank you @etshy

**Pull request merged**